packages/patternfly-3/patternfly-react/src/components/DateTimePicker/DatePicker.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl, InputGroup, Icon, OverlayTrigger, Popover } from '../../';
+import DateInput from './DateComponents/DateInput';
+import TodayButton from './DateComponents/TodayButton';
+
+class DatePicker extends React.Component {
+  formatDate = date => {
+    const { locale } = this.props;
+    const options = { year: 'numeric', month: 'numeric', day: 'numeric' };
+    if (Date.parse(date)) {
+      const parsedValue = new Date(date);
+      return parsedValue.toLocaleString(locale, options);
+    }
+    return date;
+  };
+  state = {
+    value: new Date(this.props.value),
+    tmpValue: this.formatDate(new Date(this.props.value)),
+    hiddenValue: !this.props.hiddenValue
+  };
+  setSelected = date => {
+    let newDate = new Date(this.state.value);
+    if (Date.parse(date)) {
+      newDate = new Date(date);
+    }
+    this.setState({ value: newDate, tmpValue: this.formatDate(newDate) });
+  };
+
+  render() {
+    const { value, hiddenValue } = this.state;
+    const { locale, weekStartsOn, id, placement } = this.props;
+    const popover = (
+      <Popover id={id} className="bootstrap-datetimepicker-widget date-picker-pf usetwentyfour">
+        <ul className="list-unstyled">
+          <li>
+            <DateInput date={value} setSelected={this.setSelected} locale={locale} weekStartsOn={weekStartsOn} />
+          </li>
+          <li className="picker-switch accordion-toggle">
+            <TodayButton setSelected={this.setSelected} />
+          </li>
+        </ul>
+      </Popover>
+    );
+    return (
+      <div>
+        <InputGroup className="input-group date-time-picker-pf">
+          <FormControl
+            aria-label="date-time-picker-input"
+            type="text"
+            value={hiddenValue ? '' : this.state.tmpValue}
+            onChange={e => this.setState({ tmpValue: e.target.value })}
+            onBlur={e => this.setSelected(e.target.value)}
+          />
+          <OverlayTrigger trigger="click" placement={placement} overlay={popover} rootClose>
+            <InputGroup.Addon className="date-picker-pf">
+              <Icon type="fa" name="calendar" />
+            </InputGroup.Addon>
+          </OverlayTrigger>
+
+          <InputGroup.Addon className="clear-button">
+            <Icon type="fa" name="close" onClick={() => this.setState({ hiddenValue: true, value: new Date() })} />
+          </InputGroup.Addon>
+        </InputGroup>
+      </div>
+    );
+  }
+}
+
+DatePicker.propTypes = {
+  value: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
+  locale: PropTypes.string,
+  weekStartsOn: PropTypes.number,
+  hiddenValue: PropTypes.bool,
+  id: PropTypes.string,
+  placement: PropTypes.oneOf(['top', 'bottom'])
+};
```

Comment:
Comments for prop values

Reply:
changed to the overlay proptype of placement